### PR TITLE
[BE1] add signature check for digital cash

### PR DESCRIPTION
**Pull request by MaximeZmt**
Add Signature check for back-end 1 for digital cash project.
It will do the same check then as the scala back-end.

Waiting that https://github.com/dedis/popstellar/pull/1123 is merged, then merging on master

### CODE REVIEW

File: be1-go/message/messagedata/coin_post_transaction.go [hunk 1]
```diff
@@ -66,6 +68,20 @@ func (message PostTransaction) Verify() error {
 			"encoded", message.TransactionID)
 	}
 
+	err = message.verifyTransactionId()
+	if err != nil {
+		return err
```
Comment (Robingoumaz) [inline be1-go/message/messagedata/coin_post_transaction.go:73]: You should wrap the error

File: be1-go/message/messagedata/coin_post_transaction.go [hunk 2]
```diff
@@ -66,6 +68,20 @@ func (message PostTransaction) Verify() error {
...
+	}
+
+	err = message.verifySignature()
+	if err != nil {
+		return err
```
Comment (Robingoumaz) [inline be1-go/message/messagedata/coin_post_transaction.go:78]: Same here

**Review by Robingoumaz — CHANGES_REQUESTED** [review]
Very good, just need to wrap two errors

**Review by lourkeur — APPROVED** [review]
LGTM

**Review by lourkeur — APPROVED** [review]

**Review by Robingoumaz — APPROVED** [review]

File: be1-go/message/messagedata/coin_post_transaction.go [hunk 3]
```diff
@@ -113,6 +129,50 @@ func (message PostTransaction) Verify() error {
...
+		}
+
+		publicKeySender, err := base64.URLEncoding.DecodeString(inp.Script.PubKey)
+		if err != nil {
+			return xerrors.Errorf("failed to decode public key string: %v", err)
+		}
+
+		err = schnorr.VerifyWithChecks(crypto.Suite, publicKeySender, dataBytes, signatureBytes)
+		if err != nil {
+			return xerrors.Errorf("failed to verify signature : %v", err)
```
Comment (Tesa27) [inline be1-go/message/messagedata/coin_post_transaction.go:169]: I think that a custom error from `answer` should be applied here instead of `xerrors`

Comment (MaximeZmt) [reply be1-go/message/messagedata/coin_post_transaction.go:169]: Only Here ? or for all of them ? What would concretely differ from doing that?

File: be1-go/message/messagedata/coin_post_transaction.go [hunk 4]
```diff
@@ -113,6 +129,50 @@ func (message PostTransaction) Verify() error {
 	return nil
 }
 
+func (message PostTransaction) verifySignature() error {
+	var sigComp []string
```
Comment (nkcr) [inline be1-go/message/messagedata/coin_post_transaction.go:133]: what you need here is a `bytes.Buffer`:

```go
sigComps := new(bytes.Buffer)
...
sigComps.WriteString(value)
sigComps.WriteString(hash)
...
dataBytes := sigComps.Bytes()
```

**Pull request merged**